src/System.Private.CoreLib/shared/System/Threading/ThreadLocal.cs

```diff
@@ -450,15 +450,16 @@ public IList<T> Values
         /// <summary>Gets all of the threads' values in a list.</summary>
         private List<T> GetValuesAsList()
         {
-            List<T> valueList = new List<T>();
+            LinkedSlot linkedSlot = _linkedSlot;
             int id = ~_idComplement;
-            if (id == -1)
+            if (id == -1 || linkedSlot == null)
```

Comment:
Is this fixing a bug when `_linkedSlot`? Do we need a test for it?

Reply:
Seems like a possibility that after Dispose is called and another thread calls this method, because there are no restrictions on the loads here, the other thread could read an old value for `_idComplement` (`> 0`) and a new value for `_linkedSlot` (`== null`), then it would throw `NullReferenceException` instead of `ObjectDisposedException`. I didn't see anything that would prevent that, and thought it would be difficult to repro but it seems to be fairly easy, will add a test to the CoreFX PR.